forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt

```diff
@@ -33,22 +45,80 @@ data class FROptions(val server: Server,
                     && old?.urlPath == new?.urlPath
                     && old?.logger == new?.logger
         }
+
+
     }
+
     @Throws(IllegalArgumentException::class)
     @JvmName("validateConfig")
     internal fun validateConfig() {
         require(server.url.isNotBlank()) { "AM URL cannot be blank" }
         require(server.realm.isNotBlank()) { "Realm cannot be blank" }
         require(server.cookieName.isNotBlank()) { "cookieName cannot be blank" }
     }
+
+    private fun getOkHttpClient(url: URL): OkHttpClient {
+        val networkConfig = NetworkConfig.networkBuilder()
+            .timeUnit(TimeUnit.SECONDS)
+            .host(url.authority)
+            .interceptorSupplier {
+                listOf<Interceptor>(
+                    OkHttpRequestInterceptor()
+                )
+            }
+            .build()
+
+        // Obtain instance of OkHttp client
+        val httpClient = getInstance().lookup(networkConfig)
+        return httpClient
+    }
+
+    suspend fun discover(discoverUrl: String): FROptions {
+        return withContext(Dispatchers.IO) {
+            suspendCancellableCoroutine { continuation ->
+                try {
+                    val url = URL(discoverUrl)
+                    val httpClient = getOkHttpClient(url)
+                    val request = Request.Builder()
+                        .header(ServerConfig.ACCEPT_API_VERSION, "resource=1.0")
+                        .url(discoverUrl)
+                        .get().build()
+                    httpClient.newCall(request).execute().use { response ->
+                        if (!response.isSuccessful) {
+                            throw ApiException(
+                                response.code,
+                                response.message,
+                                response.body?.string()
+                            )
+                        }
+
+                        val openIdConfiguration = response.body?.string()?.let {
+                            JSONObject(it)
+                        }
+                        this@FROptions.urlPath.authorizeEndpoint =
```

Comment:
Can we keep FROptions immutable?
Clone the existing FROptions and replace the urlPath?
Something like this:
```kotlin
 suspend fun discover(discoverUrl: String) : FROptions {
        // Fetch the new UrlPath from the discoverUrl...
        val newUrlPath = UrlPath() // Replace this with the actual UrlPath fetched from the discoverUrl

        // Create a copy of the current FROptions object and modify the urlPath
        return this.copy(urlPath = newUrlPath)
    }
```